### DISCUSSION TITLE
Initial page load or page navigation in GetInitialProps

### DISCUSSION THREAD

**Discussion by pecoram** [vercel/next.js#17425] [Help]
Hi,
Is there a way to know if it is the initial page load or a page navigation inside getInitialProps?

Thanks!

**Comment by jamesmosier** [comment]
You could check `if (context.req) { /* server */ } else { /* client */ }`

The [docs](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps) mention `req` and `res` are only available on the server, which would be an initial page load (full page refresh).

**Reply by pecoram** [reply]
Yes, I know that but the problem is that pages with getServerSideProps also have req and req and it is not possible to distinguish those loaded at the beginning from those navigated.

**Reply by jamesmosier** [reply]
Hey @pecoram! Do you have an example of the question you are asking? Your original question mentioned `getInitialProps` so that's what I answered 😃
